### DISCUSSION TITLE
Plex integration with PhotoPrism, no need to reinvent the wheel.

### DISCUSSION THREAD

**Discussion by LeehamElectronics** [photoprism/photoprism#2316] [Ideas]
I saw a few posts about making an AppleTV app for PhotoPrism, what about an iOS app, an Android app? I don't think it's necessary to make an app for this when great solutions like Plex are available. 

Here is what my setup currently looks like. I have my photo library consisting of two folders, shared, and personal. Within the shared folder there are subfolders that are categorized depending on who they are shared with, and personal contains a sub folder for each individual user's photos. Looks like this:
![image](https://user-images.githubusercontent.com/51737378/167788398-b55eacf8-2940-4f89-9225-8fe732bdcf71.png)

I have PhotoPrism looking over everything in `photo-library/` and I give each of my Plex users permission to view their associated photo libraries. This works pretty well at the moment as my family members can simply use Plex to view all their photos with ease. Soon they will also be able to use PhotoPrism when I setup multi-user capabilities _(not sure how yet but might end up having a separate instance of PhotoPrism per user?)_

Here's the downside, Plex can't really handle tagging photos, and even if they could, I would rather keep things in sync with PhotoPrisms AI tagging system. I made [this post](https://forums.plex.tv/t/tagging-photo-libraries-feature-in-plex-is-severely-limited/792382) on Plex's forums to see if they could integrate some way _(perhaps an API?)_ for other programs like PhotoPrism to upload metadata and tags into Plex's database. I hope this is possible and they work on implementing the idea, but I won't be holding my breath.

But now I have noticed that some files such as JPEG files are able to save tags and other metadata natively within the file itself, and plex actually reads that data it seems. What if, PhotoPrism edited these files and appended the tags that it generates from the AI software so that Plex can read from it? I'm not sure how to get around other file formats like PNG since they don't support that kind of metadata, but it's a start.

I feel like having this working would make Plex a much better candidate for substituting a PhotoPrism app on every single platform. There are a lot of platforms out there and Plex is pretty much available on all of them so might as well try and make the most of it.

Thanks.

**Comment by EarthBoundX5** [comment]
I'm actually not a fan of modifying source files; as it can have affects like changing the last modified date on the file metadata; which can impact how some tools view the file.

I actually put my files for Plex behind Read-Only shares for this very reason; I don't trust that Plex won't one day attempt to modify my files.

I'd much rather tools adopt a universal standard such as NFO files.  Let PhotoPrism write all it needs to for updates to the photo to a separate file.  Let Plex ingest that NFO file.  Sure, it takes work from both ends, but so would a bidirectional use of the APIs; but at least support for NFO can have far reaching resolutions.

**Reply by lastzero** [reply]
I have added the extension `.nfo` to the known file types, see https://github.com/photoprism/photoprism/commit/e182081d6b103734665d577ccbfaa10967e6302c. Unfortunately, `.nfo` files are not standardized and in most cases they are simple text files that do not contain structured metadata like XMP or ExifTool JSON files:

- https://en.wikipedia.org/wiki/.nfo
- https://www.tinymediamanager.org/docs/movies/nfo-formats

@EarthBoundX5 Feel free to provide us with specific usage examples (e.g. which metadata is important to you and how it can be found/stored/synced) and corresponding file samples for testing and future use:
- https://www.photoprism.app/contact#file-samples
- https://dl.photoprism.app/samples/Formats/Sidecar/
